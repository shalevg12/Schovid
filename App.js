import React ,{ useState } from 'react';
import { StyleSheet, Text, View } from 'react-native';


import Navigation from './navigation/Navigation';

import { LogBox } from 'react-native'; /// unfreeze for running on phones
import AsyncStorage from '@react-native-async-storage/async-storage';

LogBox.ignoreLogs(['Require cycle']); /// unfreeze for running on phones

/*const FetchFonts =()=>{
  return Font.loadAsync({
    'open-sans': require('./assets/fonts/OpenSans-Regular.ttf'),
    'open-sans-bold': require('./assets/fonts/OpenSans-Bold.ttf')
  })
}*/


export default function App() {
  const [FontLoaded,setFontLoaded] = useState(false);

  // if(!FontLoaded){
  //   return (
  //   <AppLoading
  //     startAsync={FetchFonts}
  //     onFinish={() => setFontLoaded(true)}
  //     />
  //     );
  //   } 
  AsyncStorage.clear();
  return <Navigation/>;
}

const styles = StyleSheet.create({
  screen:{ 
    flex: 1
  }
});
