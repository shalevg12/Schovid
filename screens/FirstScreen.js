import React ,{useState} from 'react'
import {View ,Alert , Text, StyleSheet, Button, TouchableWithoutFeedback, Keyboard} from 'react-native'
import Card from '../components/Card';
import { Platform } from 'react-native';
import colors from '../constants/Colors';
import Input from '../components/Input';
//import ParentSignUp from './Parent/ParentSignUp';

const FirstScreen = props => {
    return(
            <View style={styles.screen}>
                {/*<Text style={styles.title}>Welcome</Text>*/}
                <Card style={styles.inputcontainer}>
                    <View style={styles.buttoncontainer}>
                        <Button title="Sign Up" onPress={() => {
                            /*Alert.alert(
                                'Move To Parent Sign up',
                                'Your are about to move to the Parent Sign Up screen',
                                [
                                  {text: 'OK'}
                                ],
                                {cancelable: false},
                              ); */
                            props.navigation.navigate({routeName: 'Signup'});
                        }} color={colors.secondery}/>
                    </View>    
                    <View style={styles.buttoncontainer}>
                        <Button title="Sign In" onPress={() => {
                            console.log('wtf?!')
                             props.navigation.navigate({routeName: 'TypeSignin'});
                        }} color={colors.secondery} />
                    </View>
                    {/*<Input 
                        style={styles.Input}
                        blurOnSubmit
                        onChangeText={InputHandler}
                        value={enteredInput}
                        keyBoard='letters'
                    /> */}
                </Card>
            </View>
    );
};

const styles = StyleSheet.create({
    screen: {
        flex: 1,
        padding: 10,
        alignItems: 'center',
    },
    title:{
        fontSize:20,
        marginVertical: 10,
    },
    buttoncontainer:{
        flexDirection: 'row',
        width: '100%',
        justifyContent: 'center',
        paddingHorizontal: 90,
        paddingBottom: 35,
        paddingLeft: 90,
        paddingRight: 90,
        paddingTop: 30
    },
    inputcontainer:{
        width: 400,
        maxWidth: '80%',
        alignItems: 'center',
    }, 
    input: {
        width:50,
        textAlign:'center'
    }

});

export default FirstScreen;