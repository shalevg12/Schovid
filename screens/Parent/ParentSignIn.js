import React, { useCallback, useContext, useEffect, useState } from 'react';
import {View, Text, StyleSheet ,Button, Alert , TouchableOpacity , Keyboard} from 'react-native';
import { NavigationActions, StackActions } from 'react-navigation'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import colors from '../../constants/Colors';
import Input from '../../components/Input';
import { TouchableWithoutFeedback } from 'react-native-gesture-handler';
import { createParent } from '../../actions/Parent';
import Firebase ,{db} from '../../firebase/fire';
import AsyncStorage from '@react-native-async-storage/async-storage';

import Navigation from '../../navigation/Navigation';
import darkColors from 'react-native-elements/dist/config/colorsDark';


const ParentSignIn = props => {

    const [email,setEmail]= useState('');


    const EmailHandler = EmailText => {
        setEmail(EmailText.replace(/^[0-9](9,12)/))
    }
    
    const [password,setPass]= useState('');

    const PassHandler = PassText => {
        setPass(PassText)
    }

    const [verify,setverify]=useState(false);

    const AddItem = async (saveas,save) =>{
        try{
            console.log("saving to async storage: "+ save)
            await AsyncStorage.setItem(saveas,save)
        } catch (error){
            console.warn(error)
        }
    }

    return (
        //<TouchableWithoutFeedback  onPress={Keyboard.dismiss}>
            <View style={styles.InputContainer}>
                <Text>Parent Sign In Screen</Text>

                <Input 
                    testID={'email'}
                    style={styles.inputField}
                    blurOnSubmit
                    autoCorrect={false}
                    placeholder='Email'
                    keyboardType="email-address"
                    onChangeText={EmailHandler}
                    value={email}
                />
                <Input 
                    testID={'password'}
                    style={styles.inputField}
                    blurOnSubmit
                    autoCorrect={false}
                    placeholder='Password'
                    keyboardType="visible-password"
                    onChangeText={PassHandler}
                    value={password}
                    secureTextEntry={true}
                />
                <View style={styles.buttoncontainer}>
                    <Button title="Sign In" onPress={() => {
                            console.log('pressed Sign In');
                            Firebase.auth().signInWithEmailAndPassword(email, password)
                            .then((userCredential) => {
                                db.collection("Parent").where("email", "==", email).get().then(function(querySnapshot) {
                                    querySnapshot.forEach(function(doc) {
                                            console.log("name from db collection: "+doc.data().fullname)
                                            AddItem('ParentFullname',doc.data().fullname);
                                            AddItem('ParentEmail',doc.data().email)
                                            AddItem('ParentId', doc.data().id)
                                            AddItem('ParentPhone', doc.data().phonenum)
                                            props.navigation.navigate({routeName: 'ParentProfile'})
                                        }
                                )}
                                )
                            })
                            .catch((error) => {
                                Alert.alert('Error!','Please check info again!\nEmail is case sensitive')
                                console.log('Error!\nPlease check info again!\nEmail is case sensitive')
                              });
                             
          
                        }} color={colors.secondery} />
                </View>
                
                        {/* <Button title="Sign In" onPress={() => {
                            console.log("SignInWithEmailAndPassword started");
                            Firebase.auth().signInWithEmailAndPassword(email, password)
                            .then((userCredential) => {
                                // Signed in
                                var user = userCredential;
                                (user) => {
                                    if (user){
                                      console.log(user.uid);
                                    } 
                                }
                                db.collection("Parent").where("email", "==", email).get().then(function(querySnapshot) {
                                    querySnapshot.forEach(function(doc) {
                                        // .setItem('ParentEmail',doc.data().email);
                                        // .setItem('ParentFullname',doc.data().fullname);
                                        // .setItem('ParentId',doc.data().id);
                                        // .setItem('ParentPhone',doc.data().phone);
                                    },
                                )})              
                                props.navigation.navigate({routeName:'ParentProfile'})
                                Firebase.auth().getUser(uid).then((userRecord) => {
                                    // See the UserRecord reference doc for the contents of userRecord.

                                    console.log(`Successfully fetched user data: ${userRecord.toJSON()}`);

                                })
                                })
                            .catch((error) => {
                                var errorCode = error.code;
                                var errorMessage = error.message;
                                });
                        }} color={colors.secondery} /> */}
                {/* </View> */}
            </View>
        //</TouchableWithoutFeedback>
    );
};


const styles = StyleSheet.create({
    screen: {
        marginTop: 5,
        marginBottom: 10,
        width: '100%',
        //height: windowHeight /15,
        borderColor: '#acc',
        borderRadius: 3,
        borderWidth: 1,
        flexDirection: 'row',
        alignItems: 'center',
        backgroundColor: '#fff'
    },
    InputContainer: {
        padding: 10,
        flex: 1,
        fontSize: 16,
        color: '#333',
        justifyContent: 'center',
        alignItems: 'center'
    },
    inputField: {
        padding: 10,
        marginTop: 5,
        marginBottom: 10,
        //width: windowWidth /1.5,
        //height: windowHeight /15,
        fontSize: 16,
        borderRadius: 8,
        borderWidth: 1
    }
})



export const login = (email,password) => {
    Firebase.auth().signInWithEmailAndPassword(email, password)
    .then((userCredential) => {
      // Signed in
      var user = userCredential.user;
      console.log("wtf went wrong?")
      setverify=true;
      // ...
    })
    .catch((error) => {
      var errorCode = error.code;
      var errorMessage = error.message;
    });
}


export const getUser = uid => {
    return async (dispatch, getState) => {
        try {
            const user = await db
                .collection('users')
                .doc(uid)
                .get()

            dispatch({ type: LOGIN, payload: user.data() })
        } catch (e) {
            alert(e)
        }
    }
}


export default ParentSignIn;