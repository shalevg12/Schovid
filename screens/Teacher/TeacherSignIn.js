import React, { useCallback, useContext, useEffect, useState } from 'react';
import {View, Text, StyleSheet ,Button, Alert , TouchableOpacity , Keyboard , Platform} from 'react-native';
import AsyncStorage from '@react-native-async-storage/async-storage';

import { NavigationActions, StackActions } from 'react-navigation'

import colors from '../../constants/Colors';
import Input from '../../components/Input';

import { TouchableWithoutFeedback } from 'react-native-gesture-handler';
import Firebase ,{db} from '../../firebase/fire';


import { LogBox } from 'react-native'; /// unfreeze for running on phones

LogBox.ignoreLogs(['Setting a timer']); /// unfreeze for running on phones

const TeacherSignIn = props => {

    const login = () => {
        return async (dispatch, getState) => {
            try {
                const response = await Firebase.auth().signInWithEmailAndPassword(EmailInput, PassInput)
                dispatch(getUser(response.user.uid))
                props.navigation.navigate({routeName: 'TeacherLogin'});
            } catch (e) {
                alert(e)
            }
        }
    }
    
    const getUser = uid => {
        return async (dispatch, getState) => {
            try {
                const user = await db
                    .collection('Parent')

                    .doc(uid)
                    .get()
                } 
            catch (e) 
            {
                alert(e)
            }
        }
    }

    const [EmailInput,setEmail]= useState('');

    const EmailHandler = EmailText => {
        setEmail(EmailText.replace(/^(9,12)/))
    }
    
    const [PassInput,setPass]= useState('');

    const [Verified, setVerified]= useState(false);

    const VerifiedHandler = VerifyTrue => {
        setVerified(true);
    }

    const PassHandler = PassText => {
        setPass(PassText)
    }
    //const { getItem, setItem } = useAsyncStorage('@storage_key');

    const readItemFromStorage = async () => {
      const item = await getItem();
      setValue(item);
    };
  
    const writeItemToStorage = async newValue => {
      await setItem(newValue);
      setValue(newValue);
    };

    // _StoreData = async () => {
    //     try {
    //        await AsyncStorage.setItem('user',JSON.stringify(doc.data().fullname))
    //        console.log(doc.data().fullname)
    //     } catch (error) {
    //       console.log("Something went wrong", error);
    //     }
    //   }

    const AddItem = async (saveas,save) =>{
        try{
            console.log("saving to async storage: "+ save)
            await AsyncStorage.setItem(saveas,save)
        } catch (error){
            console.warn(error)
        }
    }

    return (
        //<TouchableWithoutFeedback  onPress={Keyboard.dismiss}>
            <View style={styles.InputContainer}>

                <Text>Teacher Sign In Screen</Text>


                <Input
                    testID={'email'} 
                    style={styles.inputField}
                    blurOnSubmit
                    autoCorrect={false}
                    placeholder='Email'
                    keyboardType="email-address"
                    onChangeText={EmailHandler}
                    value={EmailInput}
                />
                <Input 
                    testID={'password'}
                    style={styles.inputField}
                    blurOnSubmit
                    autoCorrect={false}
                    placeholder='Password'
                    keyboardType="visible-password"
                    onChangeText={PassHandler}
                    value={PassInput}
                    secureTextEntry={true}
                />

                <View style={styles.buttoncontainer}>
                    <Button title="Sign In" onPress={() => {
                            console.log('pressed Sign In');
                            Firebase.auth().signInWithEmailAndPassword(EmailInput, PassInput)
                            .then((userCredential) => {
                                db.collection("Teacher").where("email", "==", EmailInput).get().then(function(querySnapshot) {
                                    querySnapshot.forEach(function(doc) {
                                            console.log("name from db collection: "+doc.data().fullname)
                                            AddItem('TeacherFullname',doc.data().fullname);
                                            AddItem('TeacherEmail',doc.data().email)
                                            AddItem('TeacherId', doc.data().id)
                                            AddItem('TeacherPhone', doc.data().phonenum)
                                            props.navigation.navigate({routeName: 'TeacherProfile'})
                                            //resetStack(); //unfreeze in final
                                        }
                                )}
                                )
                            })
                            .catch((error) => {
                                Alert.alert('Error!','Please check info again!\nEmail is case sensitive')
                                console.log('Error!\nPlease check info again!\nEmail is case sensitive')
                              });
                        }} color={colors.secondery} />
                </View>
                {/* <View style={styles.buttoncontainer}>
                    <Button title="Sign In" onPress={() => {
                            console.log('pressed Sign In');
                            Firebase.auth().signInWithEmailAndPassword(EmailInput, PassInput)
                            .then((userCredential) => {
                                // Signed in
                                var user = userCredential.user;
                                Alert.alert('Logged In!','Logged as '+userCredential)
                                
                                Firebase.auth().onAuthStateChanged(
                                    authUser => {
                                        if (authUser) {
                                            db.collection('Teacher').get().then( snapshot =>{
                                                //const students = []
                                                snapshot.forEach( doc =>{
                                                    KEY = Object.keys(doc.data());
                                                    console.log("KEYS is :"+KEY);
                                                    KEY.forEach( (key_id) => {
                                                        if(key_id=='email'){
                                                            const data = doc.data()
                                                            if(doc.data().email == EmailInput){
                                                                console.log(doc.data().fullname)
                                                            }
                                                        }
                                                    const userData = snapshot;
                                                    console.log(userData)
                                                })
                                            })
                                        }
                                    )
                                }})
                            .catch((error) => {
                                var errorCode = error.code;
                                var errorMessage = error.message;
                                Alert.alert('Error!',errorMessage)
                                });
                        })}} color={colors.secondery} /> */}
                    {/* <Button title="Sign In as Child" onPress={() => { 
                        console.log('pressed Sign In as Child');
                        db.collection("Teacher").where("email", "==", EmailInput).get().then(function(querySnapshot) {
                                querySnapshot.forEach(function(doc) {
                                // .setItem('user', doc.data().fullname)
                                // .setItem('email', doc.data().email)
                                }
                            )})   
                        props.navigation.navigate({routeName: 'ChildLogin'});
                    }} color={colors.secondery}/> */}
                {/* </View> */}
            </View>
        //</TouchableWithoutFeedback>
    );
};


const styles = StyleSheet.create({
    screen: {
        marginTop: 5,
        marginBottom: 10,
        width: '100%',
        //height: windowHeight /15,
        borderColor: '#acc',
        borderRadius: 3,
        borderWidth: 1,
        flexDirection: 'row',
        alignItems: 'center',
        backgroundColor: '#fff'
    },
    InputContainer: {
        padding: 10,
        flex: 1,
        fontSize: 16,
        color: '#333',
        justifyContent: 'center',
        alignItems: 'center'
    },
    inputField: {
        padding: 10,
        marginTop: 5,
        marginBottom: 10,
        width: 250,
        //height: windowHeight /15,
        fontSize: 16,
        borderRadius: 8,
        borderWidth: 1
    }
})

export default TeacherSignIn;