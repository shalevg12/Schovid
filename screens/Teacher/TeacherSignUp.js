import React, { useCallback, useContext, useEffect, useState } from 'react';
import {View, Text, StyleSheet ,Button, Alert , TouchableOpacity , Keyboard} from 'react-native';
import colors from '../../constants/Colors';
import Input from '../../components/Input';
import AsyncStorage from '@react-native-async-storage/async-storage';

import { NavigationActions, StackActions } from 'react-navigation'
import Firebase ,{db} from '../../firebase/fire';

const TeacherSignUp = props => {
    const AddItem = async (saveas,save) =>{
        try{
            console.log("saving to async storage: "+ save)
            await AsyncStorage.setItem(saveas,save)
        } catch (error){
            console.warn(error)
        }
    }

    const signup = async() =>{ 
        try{
            const response = await Firebase.auth().createUserWithEmailAndPassword(EmailInput, PassInput)
            if (response.user.uid) {
                const user = {
                    uid: response.user.uid,
                    email: EmailInput,
                    fullname: FullnameInput,
                    phonenum: PhoneInput,
                    id:IDInput,
                    Role: 'Teacher', 
                    TeacherOfClass: null   
                }
                db.collection('Teacher')
                    .doc(FullnameInput)
                    .set(user)
                AddItem('TeacherFullname',FullnameInput);
                AddItem('TeacherEmail',EmailInput)
                AddItem('TeacherId', IDInput)
                AddItem('TeacherPhone', PhoneInput)
                props.navigation.navigate({routeName: 'TeacherProfile'});
            }

        } catch (e){
            if(e.code == 'auth/invalid-email'){
                Alert.alert("Bad Email!", e.message)
                console.log(e);
            }
            if(e.code == 'auth/email-already-in-use'){
                Alert.alert("This Email is already Registered!", "The email address is already in use by another account.")
                console.log(e);
            }
            else{
                Alert.alert(e.code,e.message)
                console.log(e);
            }

        }
    }

    const [FullnameInput,setFname]= useState('');


    const FullnameHandler = FullnameText => {
        setFname(FullnameText.replace(/[^A-Za-z]+[^A-Za-z]/))
    }


    const [EmailInput,setEmail]= useState('');


    const EmailHandler = EmailText => {
        setEmail(EmailText.replace(/^[0-9](9,12)/))
    }


    const [PhoneInput,setPhone]= useState('');


    const PhoneHandler = PhoneText => {
        setPhone(PhoneText.replace(/^[0-9](9,12)/))
    }
    

    const [PassInput,setPass]= useState('');


    const PassHandler = PassText => {
        setPass(PassText)
    }


    const [IDInput,setID]= useState('');


    const IDHandler = IDText => {
        setID(IDText.replace(/^[0-9](9,9)/))
    }
    

    const [VerifyPass, setVerifyPass] = useState ('');


    const VerifyHandler = VerifyPassText =>{
        setVerifyPass(VerifyPassText)
    }

    const resetinputHandler =() =>{
        setenteredInput=('');
    }

    const submitHandler = useCallback(() =>{
        console.log("Submit!");
    },[]);


    return (
        //<TouchableWithoutFeedback  onPress={Keyboard.dismiss}>
            <View style={styles.InputContainer}>
                <Text>Parent Sign Up Screen</Text>
                <Input
                    testID={'fullname'}
                    style={styles.inputField}
                    blurOnSubmit
                    autoCorrect={false}
                    placeholder='Full Name'
                    keyboardType="ascii-capable"
                    onChangeText={FullnameHandler}
                    value={FullnameInput}
                />
                <Input 
                    testID={'email'}
                    style={styles.inputField}
                    blurOnSubmit
                    autoCorrect={false}
                    placeholder='Email'
                    keyboardType="email-address"
                    onChangeText={EmailHandler}
                    value={EmailInput}
                />
                <Input 
                    testID={'phone'}
                    style={styles.inputField}
                    blurOnSubmit
                    autoCorrect={false}
                    placeholder='Phone'
                    keyboardType="phone-pad"
                    onChangeText={PhoneHandler}
                    value={PhoneInput}
                />
                <Input 
                    testID={'password'}
                    style={styles.inputField}
                    blurOnSubmit
                    autoCorrect={false}
                    placeholder='Password'
                    keyboardType="visible-password"
                    onChangeText={PassHandler}
                    value={PassInput}
                    secureTextEntry={true}
                />
                <Input 
                    style={styles.inputField}
                    blurOnSubmit
                    autoCorrect={false}
                    placeholder='Verify Password'
                    keyboardType="visible-password"
                    onChangeText={VerifyHandler}
                    value={VerifyPass}
                    secureTextEntry={true}
                />
                <Input
                    testID={'id'}
                    style={styles.inputField}
                    blurOnSubmit
                    autoCorrect={false}
                    placeholder='ID number'
                    keyboardType="number-pad"
                    onChangeText={IDHandler}
                    value={IDInput}
                />
                <View style={styles.buttoncontainer}>
                        <Button title="Sign Up" onPress={() => {
                            if(PassInput!=''){
                                if(VerifyPass==PassInput){
                                    submitHandler();
                                    signup();
                                }
                                else{
                                    Alert.alert(
                                        'Error',
                                        'Passwords do no match!',
                                        [
                                          {text: 'OK'}
                                        ],
                                        {cancelable: false},
                                      );
                                      console.log(VerifyPass);
                                }
                            }
                            else{
                                Alert.alert(
                                    'Error',
                                    'Please enter a Password',
                                    [
                                      {text: 'OK'}
                                    ],
                                    {cancelable: false},
                                  );
                                  console.log(VerifyPass);
                            }
                        }} color={colors.secondery} />
                </View>
            </View>
        //</TouchableWithoutFeedback>
    );
};


const styles = StyleSheet.create({
    screen: {
        marginTop: 5,
        marginBottom: 10,
        width: '100%',
        //height: windowHeight /15,
        borderColor: '#acc',
        borderRadius: 3,
        borderWidth: 1,
        flexDirection: 'row',
        alignItems: 'center',
        backgroundColor: '#fff'
    },
    InputContainer: {
        padding: 10,
        flex: 1,
        fontSize: 16,
        color: '#333',
        justifyContent: 'center',
        alignItems: 'center'
    },
    inputField: {
        padding: 10,
        marginTop: 5,
        marginBottom: 10,
        //width: windowWidth /1.5,
        //height: windowHeight /15,
        fontSize: 16,
        borderRadius: 8,
        borderWidth: 1
    }
})


export default TeacherSignUp;