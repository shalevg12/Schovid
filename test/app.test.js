import React from "react";
import renderer from "react-test-renderer";
import App from '../App';
import FirstScreen from "../screens/FirstScreen";
import TeacherSignUp from "../screens/Teacher/TeacherSignUp";
import TeacherSignIn from "../screens/Teacher/TeacherSignIn";
import ParentSignUp from "../screens/Parent/ParentSignUp";
import ParentSignIn from "../screens/Parent/ParentSignIn";
import TypeSignIn from "../screens/TypeSignin";

/*test('Open first screen', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).dispatch(FirstScreen);
});

it('Test click event', () => {
    const mockCallBack = jest.fn();

    const button = shallow((<Button onClick={mockCallBack}>Ok!</Button>));
    button.find('button').simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(1);
});*/

test('renders correctly across screens', () => {
    const tree = renderer.create().toJSON();
    expect(tree).toMatchSnapshot();
}); 

/*test('renders TypeSignIn screen', () => {
    const tree = renderer.create(<TypeSignIn/>).toJSON();
    //expect(tree.children.TypeSignIn).toMatchSnapshot();
});*/ 

let FindElement = function (tree ,element) {
    let result = undefined;
    for (node in tree.children){
        if (tree.children[node].props.testID ==  element)
            result = true;
    }
    return result;
}

//----Checks TeacherSignUp---
it('Find Element Fullname in TeacherSignUp', () =>{
    let tree =renderer.create(
        <TeacherSignUp/>).toJSON();
    
    expect(FindElement(tree, 'fullname')).toBeDefined();
})

it('Find Element Email in TeacherSignUp', () =>{
    let tree =renderer.create(
        <TeacherSignUp/>).toJSON();
    
    expect(FindElement(tree, 'fullname')).toBeDefined();
})

it('Find Element Phone in TeacherSignUp', () =>{
    let tree =renderer.create(
        <TeacherSignUp/>).toJSON();
    
    expect(FindElement(tree, 'phone')).toBeDefined();
})

it('Find Element ID in TeacherSignUp', () =>{
    let tree =renderer.create(
        <TeacherSignUp/>).toJSON();
    
    expect(FindElement(tree, 'id')).toBeDefined();
})

it('Find Element Pass in TeacherSignUp', () =>{
    let tree =renderer.create(
        <TeacherSignUp/>).toJSON();
    
    expect(FindElement(tree, 'password')).toBeDefined();
})

//----Checks TeacherSignIn---

it('Find Element Email in TeacherSignIn', () =>{
    let tree =renderer.create(
        <TeacherSignIn/>).toJSON();
    
    expect(FindElement(tree, 'email')).toBeDefined();
})

it('Find Element Pass in TeacherSignIn', () =>{
    let tree =renderer.create(
        <TeacherSignIn/>).toJSON();
    
    expect(FindElement(tree, 'password')).toBeDefined();
})

//----Checks ParentSignUp---
it('Find Element Fullname in ParentSignUp', () =>{
    let tree =renderer.create(
        <ParentSignUp/>).toJSON();
    
    expect(FindElement(tree, 'fullname')).toBeDefined();
})

it('Find Element Email in ParentSignUp', () =>{
    let tree =renderer.create(
        <ParentSignUp/>).toJSON();
    
    expect(FindElement(tree, 'fullname')).toBeDefined();
})

it('Find Element Phone in ParentSignUp', () =>{
    let tree =renderer.create(
        <ParentSignUp/>).toJSON();
    
    expect(FindElement(tree, 'phone')).toBeDefined();
})

it('Find Element ID in ParentSignUp', () =>{
    let tree =renderer.create(
        <ParentSignUp/>).toJSON();
    
    expect(FindElement(tree, 'id')).toBeDefined();
})

it('Find Element Pass in ParentSignUp', () =>{
    let tree =renderer.create(
        <ParentSignUp/>).toJSON();
    
    expect(FindElement(tree, 'password')).toBeDefined();
})

//----Checks ParentSignIn---
it('Find Element Email in ParentSignIn', () =>{
    let tree =renderer.create(
        <ParentSignIn/>).toJSON();
    
    expect(FindElement(tree, 'email')).toBeDefined();
})

it('Find Element Pass in ParentSignIn', () =>{
    let tree =renderer.create(
        <ParentSignIn/>).toJSON();
    
    expect(FindElement(tree, 'password')).toBeDefined();
})

// it('Move to Login screen', () => {
//     let FirstScreenPage = renderer.create(<FirstScreen />).getInstance();
//     FirstScreenPage.
//   });