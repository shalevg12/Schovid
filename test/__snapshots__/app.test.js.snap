// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TypeSignIn screen 1`] = `null`;

exports[`renders correctly across screens 1`] = `null`;
